src/lib/lib.jl

```diff
@@ -209,7 +209,7 @@ end
 
 @generated pair(::Val{k}, v) where k = :($k = v,)
 
-@adjoint function literal_getproperty(x, ::Val{f}) where f
+@adjoint function literal_getproperty(x, ::Val{f}, getproperty=getproperty) where f
```

Comment:
I feel this is going to lead to confusion since getproperty doesn't take a `getproperty` argument.
I thikn better is to declare a `literal_getfield` that matches `literal_getproperty`

```julia
for getfun in (:getproperty, :getfield)
    @eval @adjoint function $(Symbol(:literal_, $getfun)(x, ::Val{f}) where f
        val = $getfun(x,f)
    ...
end
```

Reply:
This was added in https://github.com/FluxML/ZygoteRules.jl/pull/13. My thinking was that it should be more backwards compatible this way, but maybe we should go with your proposal instead.